WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,618 @@
+            apost.status = previousStatus
+            
+            if let strongSelf = self {
+                contextManager.saveContext(strongSelf.managedObjectContext())
+            }
+            
+            WPError.showXMLRPCErrorAlert(error)
+        }
+    }
+    
+    func promptThatPostRestoredToFilter(filter: PostListFilter) {
+        var message = NSLocalizedString("Page Restored to Drafts", comment: "Prompts the user that a restored page was moved to the drafts list.")
+        
+        switch filter.filterType {
+        case .Published:
+            message = NSLocalizedString("Page Restored to Published", comment: "Prompts the user that a restored page was moved to the published list.")
+        break
+        case .Scheduled:
+            message = NSLocalizedString("Page Restored to Scheduled", comment: "Prompts the user that a restored page was moved to the scheduled list.")
+            break
+        default:
+            break
+        }
+        
+        let alertCancel = NSLocalizedString("OK", comment: "Title of an OK button. Pressing the button acknowledges and dismisses a prompt.")
+        
+        let alertController = UIAlertController(title: nil, message: message, preferredStyle: .Alert)
+        alertController.addCancelActionWithTitle(alertCancel, handler: nil)
+        alertController.presentFromRootViewController()
+    }
+    
+    // MARK: - Filter Related
+    
+    func keyForCurrentListStatusFilter() -> String {
+        return self.dynamicType.currentPageListStatusFilterKey
+    }
+    
+    // MARK: - Cell Delegate Methods
+    
+    func cell(cell: UITableViewCell!, receivedMenuActionFromButton button: UIButton, forProvider contentProvider: WPPostContentViewProvider!) {
+        let page = contentProvider as! Page
+        let objectID = page.objectID
+        
+        let viewButtonTitle = NSLocalizedString("View", comment: "Label for a button that opens the page when tapped.")
+        let draftButtonTitle = NSLocalizedString("Move to Draft", comment: "Label for a button that moves a page to the draft folder")
+        let publishButtonTitle = NSLocalizedString("Publish Immediately", comment: "Label for a button that moves a page to the published folder, publishing with the current date/time.")
+        let trashButtonTitle = NSLocalizedString("Move to Trash", comment: "Label for a button that moves a page to the trash folder")
+        let cancelButtonTitle = NSLocalizedString("Cancel", comment: "Label for a cancel button")
+        let deleteButtonTitle = NSLocalizedString("Delete Permanently", comment: "Label for a button permanently deletes a page.")
+        
+        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .ActionSheet)
+        alertController.addCancelActionWithTitle(cancelButtonTitle, handler: nil)
+        
+        if let filter = currentPostListFilter()?.filterType {
+            if filter == .Trashed {
```

Comment:
I wouldn't mind the change, but I'd rather avoid that in this PR as this code is going to be refactored after this PR goes through.

I'd also prefer not to add too many improvements that could cause me to break the logic.

I know it's minor, please bear with me here!  Thanks!